app/src/test/java/com/kks/nimblesurveyjetpackcompose/repo/survey/SurveyRepoImplTest.kt

```diff
@@ -0,0 +1,154 @@
+package com.kks.nimblesurveyjetpackcompose.repo.survey
+
+import com.google.common.truth.Truth.assertThat
+import com.kks.nimblesurveyjetpackcompose.attributeAnswerResponse
+import com.kks.nimblesurveyjetpackcompose.attributeQuestionResponse
+import com.kks.nimblesurveyjetpackcompose.includedAnswerResponse
+import com.kks.nimblesurveyjetpackcompose.includedQuestionResponse
+import com.kks.nimblesurveyjetpackcompose.model.QuestionDisplayType
+import com.kks.nimblesurveyjetpackcompose.model.ResourceState
+import com.kks.nimblesurveyjetpackcompose.model.request.SubmitSurveyRequest
+import com.kks.nimblesurveyjetpackcompose.model.response.BaseResponse
+import com.kks.nimblesurveyjetpackcompose.model.response.SurveyIncludedRelationshipsResponse
+import com.kks.nimblesurveyjetpackcompose.model.toSurveyQuestionRequest
+import com.kks.nimblesurveyjetpackcompose.network.Api
+import com.kks.nimblesurveyjetpackcompose.surveyAnswer
+import com.kks.nimblesurveyjetpackcompose.surveyQuestion
+import io.mockk.coEvery
+import io.mockk.coVerify
+import io.mockk.mockk
+import kotlinx.coroutines.ExperimentalCoroutinesApi
+import kotlinx.coroutines.flow.collect
+import kotlinx.coroutines.flow.last
+import kotlinx.coroutines.test.runTest
+import org.junit.Before
+import org.junit.Test
+
+@ExperimentalCoroutinesApi
+class SurveyRepoImplTest {
+
+    val api: Api = mockk()
+    lateinit var sut: SurveyRepoImpl
+
+    @Before
+    fun setup() {
+        sut = SurveyRepoImpl(api)
+    }
+
+    @Test
+    fun `When getSurveyDetail and return success`() = runTest {
+        coEvery { api.getSurveyDetail(any()) } returns BaseResponse(
+            included = listOf(
+                includedQuestionResponse,
+                includedAnswerResponse
+            )
+        )
+
+        val expected = sut.getSurveyDetails("0").last()
+
+        assertThat(expected).isInstanceOf(ResourceState.Success::class.java)
```

Comment:
We can use the `shouldBeTypeOf` extension method here 😉 Then we we won't need to use the `truth` library either 😄 
```suggestion
        expected.shouldBeTypeOf<ResourceState.Success>()
```

Reply:
@ryan-conway There is no built in `shouldBeTypeOf` extension. Do you want me to create one instead?

Reply:
I just made a quick search in your repo and it is from `Kotest`. So I think it is just a different third-party library.
```
    testImplementation("io.kotest:kotest-assertions-core:${Versions.TEST_KOTEST_VERSION}")
```

Reply:
My mistake, I thought you were already using this library 😅